homeassistant/components/met_eireann/weather.py

```diff
@@ -0,0 +1,150 @@
+"""Support for Met Éireann weather service."""
+import logging
+
+from homeassistant.components.weather import WeatherEntity
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_NAME,
+    LENGTH_METERS,
+    LENGTH_MILES,
+    PRESSURE_HPA,
+    PRESSURE_INHG,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util.distance import convert as convert_distance
+from homeassistant.util.pressure import convert as convert_pressure
+
+from .const import ATTRIBUTION, CONDITION_MAP, DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def format_condition(condition: str):
+    """Map the conditions provided by the weather API to those supported by the frontend."""
+    if condition is not None:
+        for key, value in CONDITION_MAP.items():
+            if condition in value:
+                return key
+    return condition
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add a weather entity from a config_entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        [
+            MetEireannWeather(
+                coordinator, config_entry.data, hass.config.units.is_metric, False
+            ),
+            MetEireannWeather(
+                coordinator, config_entry.data, hass.config.units.is_metric, True
+            ),
+        ]
+    )
+
+
+class MetEireannWeather(CoordinatorEntity, WeatherEntity):
+    """Implementation of a Met Éireann weather condition."""
+
+    def __init__(self, coordinator, config, is_metric, hourly):
+        """Initialise the platform with a data instance and site."""
+        super().__init__(coordinator)
+        self._config = config
+        self._is_metric = is_metric
+        self._hourly = hourly
```

Comment:
Only enable one of the entities by default.

https://developers.home-assistant.io/docs/entity_registry_disabled_by#integrations-setting-default-value-of-disabled_by-for-new-entity-registry-entries